components/org.wso2.carbon.identity.application.authenticator.biometric.device.handler/src/main/java/org/wso2/carbon/identity/application/authenticator/biometric/device/handler/impl/DeviceHandlerImpl.java

```diff
@@ -0,0 +1,155 @@
+        if (log.isDebugEnabled()) {
+            log.debug("Verifying digital signature");
+        }
+        if (!verifySignature(registrationRequest.getSignature(), registrationRequest.getPublicKey(), cacheEntry)) {
+            throw new BiometricdeviceHandlerServerException("Could not verify source");
+        }
+        device = new Device(registrationRequest.getDeviceId(), registrationRequest.getDeviceName(),
+                registrationRequest.getDeviceModel(), registrationRequest.getPushId(), registrationRequest.getPublicKey());
+        DeviceCache.getInstance().addToCacheByRequestId(new BiometricDeviceHandlerCacheKey(device.getDeviceId()),
+                new DeviceCacheEntry(device));
+        DeviceDAOImpl.getInstance().registerDevice(device);
+        return device;
+    }
+
+    @Override
+    public void unregisterDevice(String deviceId) throws BiometricdeviceHandlerServerException,
+            BiometricDeviceHandlerClientException, SQLException {
+        DeviceDAOImpl.getInstance().unregisterDevice(deviceId);
+
+    }
+
+    @Override
+    public void editDeviceName(String deviceId, String newDeviceName) throws BiometricdeviceHandlerServerException,
+            SQLException {
+        DeviceCacheEntry cacheEntry = DeviceCache.getInstance()
+                .getValueFromCacheByRequestId(new BiometricDeviceHandlerCacheKey(deviceId));
+        if (cacheEntry != null) {
+            if (cacheEntry.getDevice().getDeviceName().equals(newDeviceName)) {
+                DeviceDAOImpl.getInstance().editDeviceName(deviceId, newDeviceName);
+            }
+            DeviceCache.getInstance().clearCacheEntryByRequestId(new BiometricDeviceHandlerCacheKey(deviceId));
+        } else {
+            DeviceDAOImpl.getInstance().editDeviceName(deviceId, newDeviceName);
+        }
+    }
+
+    @Override
+    public Device getDevice(String deviceId) throws BiometricDeviceHandlerClientException, SQLException,
+            BiometricdeviceHandlerServerException, IOException {
+        return DeviceDAOImpl.getInstance().getDevice(deviceId);
+    }
+
+    @Override
+    public ArrayList<Device> lisDevices() throws BiometricdeviceHandlerServerException,
+            BiometricDeviceHandlerClientException, SQLException, UserStoreException, IOException {
+        return DeviceDAOImpl.getInstance().listDevices();
+    }
+
+    @Override
+    public DiscoveryData getDiscoveryData() {
+        if (log.isDebugEnabled()) {
+            log.debug("Retrieving data to generate QR code");
+        }
+        String deviceId = UUID.randomUUID().toString();
+        User user = getAuthenticatedUser();
+        String tenantDomain = user.getTenantDomain();
+        UUID challenge = UUID.randomUUID();
+        String registrationUrl = "https://192.168.1.6:9443/t/" +
+                user.getTenantDomain() + "/me/biometricdevice";
+        String authUrl = "https://192.168.1.6:9443/t/" + user.getTenantDomain() + "/me/biometric-auth";
```

Comment:
Shall we move "https://192.168.1.6:9443/t/" and "/me/biometric-auth" as constants to the DeviceHandlerConstants file.